### DISCUSSION TITLE
Add facet argument to annotate()

### DISCUSSION THREAD

**Discussion by joranE** [ggplot2-extenders/ggplot-extension-club#22] [Ideas]
It might be useful to have a facet argument directly available in the `annotate()` functions, to allow for placing an annotation on one or several specific facet panels.

My general thoughts are that `annotate()` would gain a `facet` argument that would accept a vector of values that correspond to values in the faceting variable.

A very simple example:

```
library(ggplot2)

p <- 
  ggplot(
  data = mtcars,
  aes(x = mpg,y = hp)
) + 
  facet_wrap(~cyl) + 
  geom_point()

# annotate() places annotation on every facet
p + 
  annotate(
    geom = "text",
    x = 20,
    y = 300,
    label = "An annotation"
  )

# The traditional way to add an annotation to a single
# facet is to use an explicit layer with the associated
# geom_*() call
p + 
  geom_text(
    data = data.frame(
      mpg = 20,
      hp = 300,
      cyl = 4,
      label = "An annotation"
    ),
    aes(label = label)
  )

# A potentially useful extension to annotate() might be 
# to include a "facet" argument
p + 
  annotate(
    geom = "text",
    x = 20,
    y = 300,
    facet = "4", # panel specified by generic "facet" argument
    label = "An annotation"
  )

# or alternatively
p + 
  annotate(
    geom = "text",
    x = 20,
    y = 300,
    cyl = "4", # panel specified using the variable name defining the facets
    label = "An annotation"
  )
```

Some complications that come to mind include how to specify panels when faceting on combinations of variables.

**Comment by teunbrand** [comment]
Interesting we were just having a similar discussion over at #16. In particular in https://github.com/teunbrand/ggplot-extension-club/discussions/16#discussioncomment-9843972 I proposed a function towards the same goal.

**Reply by EvaMaeRey** [reply]
https://mastodon.social/@joranelias/112655091808860118  Timing not totally coincidental.

**Comment by EvaMaeRey** [comment]
I think some good ideas here.   For me too, annotation in a single panel does feel like it requires a lot of concentration and focus - more than it should.  I put some ideas in #16 - wanting to try out more with compute_layer().

**Reply by EvaMaeRey** [reply]
I think  '# panel specified using the variable (category) name defining the facets' seems a bit more principled; #16 has more facet index ideas.
